evm_loader/program/src/solana_backend.rs

```diff
@@ -199,58 +107,38 @@ impl<'a> SolanaBackend<'a> {
     }
 }
 
-impl<'a> Backend for SolanaBackend<'a> {
+impl<'a, 's> Backend for SolanaBackend<'a, 's> {
     fn gas_price(&self) -> U256 { U256::zero() }
-    fn origin(&self) -> H160 { self.aliases.borrow()[1].0 }
+    fn origin(&self) -> H160 { self.account_storage.contract_id() }
```

Comment:
Here we need to return `origin` - the original sender from the transaction.
When one contract calls another, then origin stays unchanged, whereas `sender` changed to calling contract.